### PR TITLE
Feature/223 naming convention for canvas operators - identity config operator

### PR DESCRIPTION
**Pull request by LesterThomas**
This has refactored the operators as per #223 

This PR includes the refactored identity config operator for keycloak (previously called securityControllerKeycloak.

### CODE REVIEW

File: charts/canvas-oda/Chart.yaml
Comment (adarshkumar4) [inline charts/canvas-oda/Chart.yaml:1]: this [version](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/canvas-oda/Chart.yaml#L18) number has been used [previously](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/canvas-oda/Chart.yaml#L22).

File: charts/component-operator/templates/deployment.yaml
Comment (adarshkumar4) [inline charts/component-operator/templates/deployment.yaml:1]: this [port](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/identityconfig-operator-keycloak/templates/deployment.yaml#L39) and this [port](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/component-operator/templates/deployment.yaml#L29) have different names.

File: charts/component-operator/values.yaml
Comment (adarshkumar4) [inline charts/component-operator/values.yaml:1]: security-listener image, credentials and configmap should be removed as these fields have been moved to identityconfig-operator-keycloak [values.yaml](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/identityconfig-operator-keycloak/values.yaml) file.

**Comment by adarshkumar4** [comment]
- this [version](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/canvas-oda/Chart.yaml#L18) number has been used [previously](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/canvas-oda/Chart.yaml#L22).
- this [port](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/identityconfig-operator-keycloak/templates/deployment.yaml#L39) and this [port](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/component-operator/templates/deployment.yaml#L29) have different names.
- security-listener image, credentials and configmap should be removed from component-operator [values.yaml](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/component-operator/values.yaml) as these fields have been moved to identityconfig-operator-keycloak [values.yaml](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/identityconfig-operator-keycloak/values.yaml) file.
- [controllerRole](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/oda-crds/templates/oda-identityconfig-crd.yaml#L27) is repeated twice in crds.
- In some instances, the term `seccon` is replaced with `idconfop` which stands for `Identity Config Operator`, while in other cases, it is replaced with `idkop` which stands for `Identity Kubernetes Operator`. It is essential to define the purpose of both terms and make corresponding changes to the relevant instances based on that decision.

**Comment by LesterThomas** [comment]
> * In some instances, the term `seccon` is replaced with `idconfop` which stands for `Identity Config Operator`, while in other cases, it is replaced with `idkop` which stands for `Identity Kubernetes Operator`. It is essential to define the purpose of both terms and make corresponding changes to the relevant instances based on that decision.

I think we should use the term `idkop` as there will be other Identity operators (e.g. the Identity operator for EntraDI).

At present we are creating the system user (that the Canvas user uses to call API's on components) `idconfop` - this system user is created by the Identity Config operator, but it servers a wider purpose. I propose we call this `CanvasSystemUser`

**Comment by LesterThomas** [comment]
> * this [port](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/identityconfig-operator-keycloak/templates/deployment.yaml#L39) and this [port](https://github.com/tmforum-oda/oda-canvas/blob/c30c51ded35303e774d58213e5d654c561c06a86/charts/component-operator/templates/deployment.yaml#L29) have different names.

Good spot - the second port on the component-operator is not needed (it is from when we packaged the identity listener with the component operator). As a default, most operators don't need to expose a port (it is just the identity listener at the moment). 
I've removed the port from the component-operator.

**Pull request merged**